geonode/api/resourcebase_api.py

```diff
@@ -481,6 +536,20 @@ def create_response(
             content_type=build_content_type(desired_format),
             **response_kwargs)
 
+    def get_resource_type(self, obj):
```

Comment:
Could we reuse this logic here https://github.com/GeoNode/geonode/blob/master/geonode/base/templatetags/base_tags.py#L125